src/Widgets/TerminalWidget.vala

```diff
@@ -608,6 +616,15 @@ namespace Terminal {
             Posix.kill (child_pid, Posix.Signal.TERM);
             reset (true, true);
             active_shell (old_loc);
+            check_cwd_changed ();
```

Comment:
if you call check_cwd_changed in active_shell you can remove it here.

Reply:
Well spotted.